writers/mongodb/init.go

```diff
@@ -0,0 +1,30 @@
+package mongodb
+
+import (
+	"time"
+
+	"gopkg.in/mgo.v2"
+)
+
+// Connect creates a connection to the MongoDB instance. A non-nil error
+// is returned to indicate failure.
+func Connect(addr string, tout int, socketTout int, db string,
```

Comment:
As I said before, it's perfectly acceptable to define two timeout constants (or one) for socket and connection timeouts, and use it directly rather than passing it as factory parameter.